gui/chiaki.appdata.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2020 Florian Märkl <chiaki@metallic.software> -->
+<component type="desktop">
+  <id>com.metallic.chiaki</id>
```

Comment:
Sorry, but this is incorrect and will cause serious issues! GNOME Software and other utilities won't be able to find the desktop file. Either change the desktop, icon and appstream file names to `com.metallic.chiaki` prefix as well (+ change the id to `com.metallic.chiaki.desktop`) or change it back to `chiaki.desktop`.

Regarding the Flatpak repository, it's unfortunately too late, the repo was already [created](https://github.com/flathub/com.github.thestr4ng3r.Chiaki) under the `com.github.thestr4ng3r.Chiaki` (app id in the Flatpak AppStream metadata is the same) and it is unfortunately not possible to change this without EOLing the whole repo and creating it again. :-/

Reply:
/cc @thestr4ng3r

Reply:
`com.github` makes no sense at all, I don't work for GitHub. The id on Android is `com.metallic.chiaki` so it should be consistent.
About the `chiaki.desktop`, do you have examples of other applications that use a similar ID? What I read is https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic and it clearly states that ids must follow reverse-DNS scheme.

Reply:
Another solution is to keep the id as it is, rename the appstream metadata file to `com.metallic.chiaki.appdata.xml` and adding the following line to this file: `<launchable type="desktop-id">chiaki.desktop</launchable>`

It should work, but I still don't recommend it. The best solution, at least in my opinion, is to revert the id back to `chiaki.desktop`.

Reply:
> Another solution is to keep the id as it is, rename the appstream metadata file to `com.metallic.chiaki.appdata.xml` and adding the following line to this file:
> 
> `<launchable type="desktop-id">chiaki.desktop</launchable>`
> 
> It should work, but I still don't recommend it. The best solution, at least in my opinion, is to revert the id back to `chiaki.desktop`.

That sounds like a perfect solution to me. What is the downside? Would you like to make another pr with this change?

Reply:
App ids **should** follow reverse-DNS scheme, but it is not required. Another freedesktop specification also specifies that desktop/icon names should also follow reverse-DNS scheme, but almost nobody actually do this. :-)

Reply:
> `com.github` makes no sense at all, I don't work for GitHub. The id on Android is `com.metallic.chiaki` so it should be consistent.

We regularly use this on Flathub for projects that are not hosted (do not have website) on any other domain, just the Git repo. It is not about you working for GitHub, but having this app hosted on GitHub. When I was creating the Flathub PR, I was unfortunately not aware about the `metallic.software` domain and as I said, I can't unfortunately change it now. :-( I will try asking Flathub admins tomorrow, but I am afraid that changing it would require EOLing the current repo and sending a new PR under a new canonical name.

> About the `chiaki.desktop`, do you have examples of other applications that use a similar ID? What I read is https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic and it clearly states that ids must follow reverse-DNS scheme.

App ids should follow reverse-DNS scheme, but it is not required. Another freedesktop specification also specifies that desktop/icon names should also follow reverse-DNS scheme, but almost nobody actually do this. :-)

Reply:
> > Another solution is to keep the id as it is, rename the appstream metadata file to `com.metallic.chiaki.appdata.xml` and adding the following line to this file:
> > `<launchable type="desktop-id">chiaki.desktop</launchable>`
> > It should work, but I still don't recommend it. The best solution, at least in my opinion, is to revert the id back to `chiaki.desktop`.
> 
> That sounds like a perfect solution to me. What is the downside? Would you like to make another pr with this change?

Well, it *should* work. But ideally, the AppStream metadata file name and app id should be the same as the desktop/icon file name. But yeah, it will hopefully work.

Reply:
Ok, for the appstream metadata I think it will be best to use the `<launchable type="desktop-id">...` solution as you suggested.

About the Flathub id, if you manage to get it changed it it would be great, but if not that is not a big problem from my side since I never use Flatpaks anyway, I just personally find these github ids quite ugly and misleading. I might for example switch to GitLab or something at some point.

Reply:
> About the Flathub id, if you manage to get it changed it it would be great, but if not that is not a big problem from my side since I never use Flatpaks anyway, I just personally find these github ids quite ugly and misleading. I might for example switch to GitLab or something at some point.

Thanks! Yeah, I don't like them as well, to be honest, but many app developers don't have any websites that we could use in app ids, just the GitHub/GitLab repos.

Reply:
I have noticed one more issue. The rDNS id is `com.metallic.chiaki`, but your domain is actually `metallic.software`, not `metallic.com`.

Reply:
Yeah, I've actually used `com.metallic` ids long before I owned this specific `metallic.software` domain (My old one is `metallic-entertainment.com`). As far as I know it's usually not required to own the real domain for these ids, but a suggestion. `software.metallic` also seems a bit ugly to me, so I prefer sticking with the `com.metallic.chiaki` one unless owning the domain is strictly required.

Reply:
#343

Reply:
As far as I know, it actually is required to own the domain used in rDNS (it is clearly noted somewhere in the freedesktop documentation) and it is definitely required on Flathub.